import "basic.proto";
import "user.proto";
import "action.proto";
import "message.proto";

enum PBResultCode{
    UNKNOW_ERROR = -1;

    SUCCESS = 0; 
    SYSTEM_ERROR = 1;
    PARSE_PB_ERROR = 2;
    PARAMETER_ERROR = 3;
    AUTH_ERROR = 4;
    NETWORK_ERROR = 5;
    TIMEOUT_ERROR = 6;

// User error code: from 10000 to 19999

    USER_NOT_EXISTS_ERROR = 10001;
    PASSWORD_INCORRECT_ERROR = 10002;
    USER_STATUS_EXCEPTION_ERROR = 10003;

// Action error code: from 20000 to 29999

    ACTION_NOT_EXISTS_ERROR = 20001;
    ACTION_STATUS_EXCEPTION_ERROR = 20002;
}



message PBResponse
{
    required PBResultCode code = 1;
    optional string error_message = 2;
    
    repeated PBAction actions = 10;
    repeated PBMessage messages = 11;
    repeated PBUser users = 12;
    repeated PBActivity activitys = 13;
    repeated PBComment comments = 14;

    optional PBUser user = 50;
    optional PBAction action = 51;
    optional PBMerchant merchant = 52;

}
